src/imitation/util/rollout.py

```diff
@@ -117,32 +130,68 @@ def finish_trajectory(self, key: Hashable = None) -> Trajectory:
       for key, array in part_dict.items():
         out_dict_unstacked[key].append(array)
     out_dict_stacked = {
-        key: np.stack(arr_list, axis=0)
-        for key, arr_list in out_dict_unstacked.items()
+      key: np.stack(arr_list, axis=0)
+      for key, arr_list in out_dict_unstacked.items()
     }
     traj = Trajectory(**out_dict_stacked)
+    assert traj.rews.shape[0] == traj.acts.shape[0] == traj.obs.shape[0] - 1
     return traj
 
-  def add_step(self, step_dict: Dict[str, np.ndarray], key: Hashable = None):
-    """Add a single step to the partial trajectory identified by `key`.
+  def add_steps_and_auto_finish(self,
```

Comment:
This method is coupled to `VecEnv` from stable baselines, but the rest of the class is not. Perhaps it would be better to have this as a standalone function?

Reply:
Don't agree with this point, as I think that `VecEnv` is a common use case throughout this repository and this file, and this method (or otherwise copying the code inside the method) is necessary for using VecEnvs with `TrajectoryAccumulator`.